source/AI.cpp

```diff
@@ -1467,8 +1469,8 @@ void AI::MoveIndependent(Ship &ship, Command &command) const
 
 		vector<int> systemWeights;
 		int totalWeight = 0;
-		const set<const System *> &links = ship.Attributes().Get("jump drive")
-			? origin->JumpNeighbors(ship.JumpRange()) : origin->Links();
+		const set<const System *> &links = ship.JumpNavigation().HasJumpDrive()
```

Comment:
Same idea as before.
This used to be fine because all we were getting was a dictionary and not a class (still could have been improve by having the Attributes() request one argument but not the point here)
Maybe we could have ship return the JumpType he has access to, by doing the same kind we already do with the ShipEvents, and adding them on top of each others if the ship has multiple kinds of jumps possible, with | etc?